### ISSUE TITLE
[FEA][QST] Update RAPIDS XGBoost to newest version

### ISSUE THREAD

**Issue by EmperorTransisthor** [rapidsai/xgboost-feedstock#45]
Hi,
When will RAPIDS XGBoost updated to version 2.0.0+? Is there any particular RAPIDS version you want to release it with?

Best wishes

**Comment by jakirkham** [comment]
There are now packages on [the `rapidsai-nightly` channel]( https://anaconda.org/rapidsai-nightly ). Please feel free to try them out and let us know what you think

**Comment by EmperorTransisthor** [comment]
I have just created new conda env using rapids 24.02a nightly release, however conda installed XGBoost 1.7.6 and I'm a little bit confused, why is that (channel rapidsai-nightly has indeed 2.0.3 available on webpage)? I have been trying now to install XGBoost 2.0.3 directly, however I only succeeded after... conda update? conda packages update? I'm not sure. My steps:

```
$ conda create --solver=libmamba -n rapids-24.02 -c rapidsai-nightly -c conda-forge -c nvidia      rapids=24.02 python=3.10 cuda-version=12.0
$ conda activate rapids-24.02
$ conda xgboost list
```
![obraz](https://github.com/rapidsai/xgboost-feedstock/assets/48327929/96e1d0c9-ff7d-4ca8-b61e-894474830ae7)

```
$ conda install rapidsai-nightly::xgboost
$ conda list xgb # same result, xgb=1.7.6

$ conda install xgboost=2.0.3 -c rapidsai-nightly
```

> Found conflicts! Looking for incompatible packages.
> This can take several minutes.  Press CTRL-C to abort.
> failed
> UnsatisfiableError: The following specifications were found to be incompatible with each other:
> 
> Output in format: Requested package -> Available versionsThe following specifications were found to be incompatible with your system:
> 
>   - feature:/linux-64::__glibc==2.35=0
>   - python=3.10 -> libgcc-ng[version='>=11.2.0'] -> __glibc[version='>=2.17']
> 
> Your installed version is: 2.35

And at this point I got a little bit confused. I have decided to update conda and all packages and see, if it allows me to install XGB 2.0.3

```
$ conda update -n base -c defaults conda
$ conda update --all -c rapidsai-nightly
$ conda install xgboost=2.0.3 -c rapidsai-nightly
```

Succeeded.


Correct me if I'm wrong, but it seems like default rapids command to setup environment, doesn't install newest versions of packages, even though it is nightly release. Is it expected behavior? I have attached console logs, if you want to check it further.

> Please feel free to try them out and let us know what you think

I'm writing master thesis and few models are trained using XGBoost. Currently, I'm working on performance tests for XGBoost devices + how Rapids Memory Manager affects performance. Is that an input you would be interested in?

**Comment by jakirkham** [comment]
Thanks Michał! 🙏

The install issue might be fixed with PR ( https://github.com/rapidsai/integration/pull/693 ). More generally we are working to get XGBoost 2.0.3 included with RAPIDS 24.02, which is [due out early February]( https://docs.rapids.ai/maintainers )

Though if you have something working for you, please don't feel a need to change it

In terms of feedback, was mostly looking for do things run ok or do you see errors that we should dig into

That said, there is definitely general interest in RAPIDS to see what you learn from performance benchmarks. Please keep us updated

**Comment by EmperorTransisthor** [comment]
Thanks :)

I believe I found one bug, but I will open separate issue for that, since it is not about updating XGBoost to newer version.
When it comes to RAPIDS, I will do my best to post in official XGBoost repo results, but since I'm short on schedule, I can't assure you if I post it very soon (tbh, XGBoost with RMM is performing way below my expectations, and I'm quite convinced I'm using it not the right way, which I need to investigate on a later date). Closing.

Kind regards
